h3/src/client/connection.rs

```diff
@@ -142,7 +144,19 @@ where
             extensions,
             ..
         } = parts;
-        let headers = Header::request(method, uri, headers, extensions)?;
+
+        //= https://www.rfc-editor.org/rfc/rfc9114#section-4.2
+        //# Characters in field names MUST be
+        //# converted to lowercase prior to their encoding.
+        let mut lower_headers = HeaderMap::new();
+
+        for (n, v) in headers.iter() {
+            lower_headers.append(
+                HeaderName::from_lowercase(n.as_str().to_ascii_lowercase().as_bytes()).unwrap(),
```

Comment:
All `HeaderName`s store themselves as lowercase. You don't need to convert them here. The `from_lowercase` only makes sure the bytes were lowercase originally, vs `from_bytes` will allow uppercase but convert it.